scripts/update_tags_per_languages.py

```diff
@@ -0,0 +1,365 @@
+    """
+
+    # language of the tags_field_string is the same as the language 
+    # of the current tag that we want to remove, 
+    # it will not be prefixed by the language.
+    if tags_field_lc == current_tag[:2]:
+        current_tag = current_tag.split(':')[1]
+    # same if new tag is the same as the language
+    if tags_field_lc == updated_tag[:2]:
+        updated_tag = updated_tag.split(':')[1]
+    
+    # convert into list to better handle upper and lower cases, split and concatenation, spaces
+    tags_fields = tags_field_string.split(",")
+    # can contain upper case letters
+    # create new list list as lower case - and remove space after commas (strip) - to get the index
+    tags_fields_lower = [x.lower().strip().replace(" ", "-", -1) for x in tags_fields]
+
+    # old tag is still in the field
+    if current_tag in tags_fields_lower:
+        index = tags_fields_lower.index(current_tag)
+        # updated tag is not yet in the field
+        # replace current tag by updated tag
+        if updated_tag not in tags_fields_lower:
+            # add space if the tag is not the first one in the string "do-not-add-space, add-space"
+            if index != 0:
+                tags_fields[index] = f" {updated_tag}"
+            else:
+                tags_fields[index] = f"{updated_tag}"
+        # updated tag is already in the field
+        # delete only instead of updating
+        else:
+            del tags_fields[index]
+    # old tag is not in the field
+    else:
+        # updated tag is not yet in the field
+        # add updated tag
+        if updated_tag not in tags_fields_lower:
+            # add space if the tag is not the first one in the string "do-not-add-space, add-space"
+            if tags_field_string != "":
+                tags_fields.append(f" {updated_tag}")
+            else:
+                tags_fields.append(f"{updated_tag}")
+        # final case, current tag missing and updated tag already in the field
+        # is equivalent to leave the field as is
+
+    tags_field_string = ",".join(tags_fields)
+    
+       
+    return tags_field_string
+
+
+def main():
+    for plural, singular in map_tags_field_url_parameter.items():
+        # 0) set variables
+        tags_list_url = f"https://{user}world.openfoodfacts.{env}/{plural}.json"
+        
+        # by default the query return 24 results. 
+        # increase to 1000 (so far chips in EN (should be crisps in EN)
+        #   had max number of products for categories with ~550)
+        products_list_for_tag_url = f"https://{user}world.openfoodfacts.{env}/{singular}/{{tag_id_placeholder}}.json?page_size=1000"
+        
+        file = os.path.abspath(os.path.join(os.path.dirname( __file__ ), '..', f'taxonomies/{plural}.txt')) 
+        
+        # country is needed otherwise <plural>_lc will be "en"
+        post_call_url = f"https://{user}{{country}}.openfoodfacts.{env}/cgi/product_jqm2.pl"
```

Comment:
(not to change) Just in case you care typically this kind of globals would better fits being attributes of a class, where you then call a method to process one tag.  

class ProcessTag:

   def __init__(self, plural, singular):
       self.tags_list_url = …
       ...
   def process(self):
       all_tags = self.get_from_api()
       etc.